### PR TITLE
Refactored cosmos element serializer to remove v3 serializer dependency

### PR DESCRIPTION
**Pull request by j82w**
# Pull Request Template

## Description

The v3 serializer code has public types which get leaked in the query core which is used in the v2 SDK. This removes those dependencies, and prevents the v2 SDK from needing to build the Cosmos Serializer code.

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### CODE REVIEW

File: Microsoft.Azure.Cosmos/src/CosmosElements/CosmosElementSerializer.cs [hunk 1]
```diff
@@ -24,12 +24,12 @@ static class CosmosElementSerializer
...
         /// <param name="memoryStream">The memory stream response from Azure Cosmos</param>
         /// <param name="resourceType">The resource type</param>
-        /// <param name="cosmosSerializationOptions">The custom serialization options. This allows custom serialization types like BSON, JSON, or other formats</param>
+        /// <param name="createJsonNavigator">The custom jsonNavigator. This allows custom serialization types like BSON, JSON, or other formats</param>
         /// <returns>Returns a memory stream of cosmos elements. By default the memory stream will contain JSON.</returns>
         internal static CosmosArray ToCosmosElements(
             MemoryStream memoryStream,
             ResourceType resourceType,
-            CosmosSerializationFormatOptions cosmosSerializationOptions = null)
+            CosmosSerializationFormatOptions.CreateCustomNavigator createJsonNavigator)
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos/src/CosmosElements/CosmosElementSerializer.cs]: How about default here => below if-else will be void then

Comment (kirankumarkolli) [reply Microsoft.Azure.Cosmos/src/CosmosElements/CosmosElementSerializer.cs]: Or another option is to ensure that CosmosSerializationFormatOptions.CreateCustomNavigatorCallback is always set to either custom or default.

---
In reply to: [328607158](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/852#discussion_r328607158) [](ancestors = 328607158)

File: Microsoft.Azure.Cosmos/src/Query/v3Query/QueryResponse.cs [hunk 2]
```diff
@@ -49,11 +50,27 @@ internal QueryResponse()
...
-        public override Stream Content => CosmosElementSerializer.ToStream(
-            this.QueryHeaders.ContainerRid,
-            this.CosmosElements,
-            this.QueryHeaders.ResourceType,
-            this.CosmosSerializationOptions);
+        public override Stream Content
+        {
+            get
+            {
+                IJsonWriter jsonWriter = null;
```
Comment (bchong95) [inline Microsoft.Azure.Cosmos/src/Query/v3Query/QueryResponse.cs]: don't set to default of T. instead use an else.

**Comment by j82w** [comment]
No longer relevant since it no longer needs to work in the v2 SDK.

**Pull request closed without merging**